src/Authenticator.js

```diff
@@ -1,56 +1,59 @@
-import JwtHelpers from './JwtHelpers';
 import { generateXSRF } from './xsrfHelpers';
-import bcrypt from 'bcrypt-nodejs';
-import http from 'http';
-import { MakeAttachUser, MakeCheckXSRF, MakeCheckAuth } from './augmentRequest';
+import { comparePass, hashPass } from './passwordHelpers';
 
-export default class Authenticator {
-  constructor({ secret, useXsrf = true, authRules = [] }) {
-    this.jwtHelpers = new JwtHelpers(secret);
-    this.useXsrf = useXsrf;
-    this.authRules = authRules;
+export default function Authenticator({ generateJWT }, createUser, fetchUser, useXsrf = true) {
+  
+  function makeTokens(user) {
+    // Generate XSRF
+    const { secret = null, token = null } = useXsrf ? generateXSRF() : {};
 
-    this.augmentRequest();
+    return {
+      user,
```

Comment:
this is optional / nitpicky.... But you could give the user object a named index to separate out the user data, and also follow the same pattern as the other ones. 
```
{
  user: user,
  JWT: generateJWT(user, secret),
  XSRF: token
}
```